src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -178,17 +178,23 @@ public int getPassCount()
   private Map<String, String> buildVariables;
   private Set<String> sensitiveBuildVariables;
   private TestData testResults = null;
-
-  // Freestyle project build
+  
   public BuildData(AbstractBuild<?, ?> build, Date currentTime, TaskListener listener) {
+    this(build, currentTime, listener, null);
+  }
+  // Freestyle project build
+  public BuildData(AbstractBuild<?, ?> build, Date currentTime, TaskListener listener, Map<String, String> additionalParams) {
     initData(build, currentTime);
 
     // build.getDuration() is always 0 in Notifiers
     rootProjectName = build.getRootBuild().getProject().getName();
+    additionalParams = additionalParams == null ? new HashMap<String, String>() : additionalParams;
     rootFullProjectName = build.getRootBuild().getProject().getFullName();
     rootProjectDisplayName = build.getRootBuild().getDisplayName();
     rootBuildNum = build.getRootBuild().getNumber();
     buildVariables = build.getBuildVariables();
+    buildVariables = new HashMap<String,String>(buildVariables);
+    buildVariables.putAll(additionalParams);
```

Comment:
why add it to build variables?

Reply:
While, I thought the most natural place of additional parameters, such as environment variables not already included, is in the build variables. Do you disagree or have a better suggestion?

Reply:
Why not just pass it as a separate field in the output?

Reply:
Fair enough